test/test_tensorclass_nofuture.py

```diff
@@ -533,20 +729,29 @@ class MyDataParent:
     X_clone = X.clone()
     td = TensorDict({}, batch_size)
     td_clone = td.clone()
-    data_nest = MyDataNest(X=X, batch_size=batch_size)
-    data = MyDataParent(X=X, y=data_nest, z=td, W=W, batch_size=batch_size)
+    data_nest = MyDataNest(X=X, v="test_nested", batch_size=batch_size)
+    v = "test_tensorclass"
+    data = MyDataParent(X=X, y=data_nest, z=td, W=W, v=v, batch_size=batch_size)
     data_nest_clone = data_nest.clone()
     assert type(data_nest_clone) is type(data_nest)
     data.y = data_nest_clone
-    assert data.tensordict["y"] is not data_nest.tensordict
-    assert data.tensordict["y"] is data_nest_clone.tensordict, (
-        type(data.tensordict["y"]),
+    assert data.y.tensordict is not data_nest.tensordict
+    assert data.y.tensordict is data_nest_clone.tensordict, (
+        type(data.y.tensordict),
         type(data_nest.tensordict),
     )
     data.X = X_clone
     assert data.tensordict["X"] is X_clone
```

Comment:
why are we checking this through the tensordict member?

BTW why tensordict is not protected? (i.e. with a `_` prefix)?